CMakeTests/AddCompileDefinitions.cmake

```diff
@@ -0,0 +1,42 @@
+# Add a C or C++ compile definitions to the current scope
+#
+# add_compile_definitions(def [def ...] [DEBUG | RELEASE])
+#
+# Can optionally add the definitions to Debug or Release configurations only, use this when
+# targeting multi-configuration generators like Visual Studio or Xcode.
+# Release configurations means NOT Debug, so it will work for RelWithDebInfo or MinSizeRel too.
+# The definitions are added to the COMPILE_DEFINITIONS folder property.
+# Supports generator expressions, unlike add_definitions()
+#
+# Examples:
+#   add_compile_definitions(FOO) -> -DFOO
+#   add_compile_definitions(_DEBUG DEBUG) -> -D_DEBUG
+#   add_compile_definitions(NDEBUG RELEASE) -> -DNDEBUG
+#   add_compile_definitions($<$<COMPILE_LANGUAGE:C>:THISISONLYFORC>)
+
+function(add_compile_definitions def)
+  set(defs)
+  list(APPEND defs ${def} ${ARGN})
+
+  list(GET defs -1 last_def)
+  if(last_def STREQUAL "DEBUG")
+    set(ADDDEF_DEBUG ON)
+    list(REMOVE_AT defs -1)
+  elseif(last_def STREQUAL "RELEASE")
+    set(ADDDEF_RELEASE ON)
```

Comment:
You could cut this down even more by doing always using a generator expression, something like (including my other suggestions):

```cmake
function(add_compile_definitions)
  set(genexp_config_test "1")
  if(ARGN MATCHES ";DEBUG_ONLY")
    set(genexp_config_test "$<CONFIG:Debug>")
    list(REMOVE_AT ARGN -1)
  elseif(ARGN MATCHES ";RELEASE_ONLY")
    set(genexp_config_test "$<NOT:$<CONFIG:Debug>>")
    list(REMOVE_AT ARGN -1)
  endif()

  set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS
    "$<${genexp_config_test}:${ARGN}>")
endfunction()
```